server/bundler/config-flag-plugin.js

```diff
@@ -0,0 +1,50 @@
+const BasicEvaluatedExpression = require( 'webpack/lib/BasicEvaluatedExpression' );
+
+// Check that the given call expression is `config.isEnabled( 'flag' )`
+// and return the `flag` literal value.
+const isConfigIsEnabledExpr = expr =>
```

Comment:
As far I understand, this checks for any literal `config.isEnabled( ... )` expressions in the source, right? I see a few potential problems with this. In decreasing order of risk:
- You could be missing some of these if only `isEnabled` gets imported. There are multiple instances of `import { isEnabled } from 'config'` in the code, for which the corresponding `isEnabled( ... )` will be missed, as far as I understand? Or is there a transformation applied before this that solves that?
- You could be missing some of these if `config` gets renamed: `import foo from 'config'`. That doesn't appear to be the case anywhere in the code at the moment, though. There may also be a previous transformation that avoids this.
- You could incorrectly be applying this to an entirely different `config` object that also happens to have an `isEnabled` method. Unlikely, but possible.

Is it possible to handle this at a more semantic level to avoid these issues?

Reply:
You're right @sgomes, all these examples could either disable the optimization, or even worse, trick us into doing in incorrectly.

I'll try to do the check more precisely: look for imports from the `config` module and check the `isEnabled` calls in any form. But I don't promise anything :)

Reply:
If that turns out to be impossible, I'd be ok with a custom lint rule that enforces the `config.isEnabled` format :)